### PR TITLE
[INLONG-10729][Sort] Sorstandalone EsSink support transform

### LINKED ISSUE #10729: [Feature][Sort] Sorstandalone EsSink support transform
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
Fixes #10729 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                .collect(Collectors.toMap(FieldInfo::getName, FieldInfo::getConverter));
+    }
+
+    @Override
+    public Map<String, Object> encode(SinkData sinkData, Context context) {
+        Map<String, Object> esMap = new HashMap<>();
+        for (FieldInfo fieldInfo : sinkInfo.getFields()) {
+            String fieldName = fieldInfo.getName();
+            String strValue = sinkData.getField(fieldName);
+            TypeConverter converter = converters.get(fieldName);
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:51]: TypeConverter may be null, right?

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/pojo/EsMapSinkInfo.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.Data;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@SuperBuilder
+@Data
+public class EsMapSinkInfo extends SinkInfo {
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/pojo/EsMapSinkInfo.java:31]: Does this map SinkInfo only work for es?

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/converter/TypeConverter.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.converter;
+
+public interface TypeConverter {
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/converter/TypeConverter.java:20]: plz add java doc for interface and its methods

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+public class EsMapSinkEncoder implements SinkEncoder<Map<String, Object>> {
+
+    private final EsMapSinkInfo sinkInfo;
+    private final Map<String, TypeConverter> converters;
+
+    public EsMapSinkEncoder(EsMapSinkInfo sinkInfo) {
+        this.sinkInfo = sinkInfo;
+        this.converters = sinkInfo.getFields()
+                .stream()
+                .collect(Collectors.toMap(FieldInfo::getName, FieldInfo::getConverter));
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:42]: Need to add a filter, if the converter is null.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public Map<String, Object> encode(SinkData sinkData, Context context) {
+        Map<String, Object> esMap = new HashMap<>();
+        for (FieldInfo fieldInfo : sinkInfo.getFields()) {
+            String fieldName = fieldInfo.getName();
+            String strValue = sinkData.getField(fieldName);
+            TypeConverter converter = converters.get(fieldName);
+            try {
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:52]: It is better to add a notnull check.
if (converter == null) {
 continue;
}

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:42]: fixed, thx

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:51]: fixed, thx

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java (hunk 5 above)
Comment (vernedeng) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/EsMapSinkEncoder.java:52]: fixed, thx

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/pojo/EsMapSinkInfo.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/pojo/EsMapSinkInfo.java:31]: fixed, rename to MapSinkInfo and MapSinkEncoder

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/converter/TypeConverter.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/converter/TypeConverter.java:20]: fixed, thx

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/DefaultEvent2IndexRequestHandler.java [hunk 6]
```diff
@@ -105,6 +110,32 @@ public EsIndexRequest parse(EsSinkContext context, ProfileEvent event) {
...
+        if (processor == null) {
+            log.error("find no any transform processor for es sink");
+            return null;
+        }
+
+        String uid = event.getUid();
+        EsIdConfig idConfig = context.getIdConfig(uid);
+        String indexName = idConfig.parseIndexName(event.getRawLogTime());
+        byte[] bodyBytes = event.getBody();
+        String strContext = new String(bodyBytes, Charset.defaultCharset());
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/DefaultEvent2IndexRequestHandler.java]: fixed, thx.

**Review by luchunliang — APPROVED** [review]

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsIdConfig.java [hunk 7]
```diff
@@ -86,6 +88,7 @@ public static EsIdConfig create(DataFlowConfig dataFlowConfig) {
                 .separator(sinkConfig.getSeparator())
                 .indexNamePattern(sinkConfig.getIndexNamePattern())
                 .fieldList(fields)
+                .charset(Charset.forName(sinkConfig.getEncodingType()))
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsIdConfig.java:91]: It maybe throw Exception for Charset.forName.
It is better to use default charset.

Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/elasticsearch/EsIdConfig.java]: fixed, thx

**Review by luchunliang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**